resources/http.go

```diff
@@ -0,0 +1,258 @@
+// Mgmt
+// Copyright (C) 2013-2017+ James Shubin and the project contributors
+// Written by James Shubin <james@shubin.ca> and the project contributors
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+package resources
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"net/http"
+	"sync"
+	"time"
+)
+
+func init() {
+	RegisterResource("http", func() Res { return &HTTPRes{} })
+}
+
+// HTTPRes is an http resource for retrieving data over http
+type HTTPRes struct {
+	BaseRes        `yaml:",inline"`
+	URL            string            `yaml:"url"`
+	Timeout        int               `yaml:"timeout"` // the http timeout in seconds
+	Method         string            `yaml:"method"`
+	Header         map[string]string `yaml:"header"`
+	Body           string            `yaml:"body"`
+	responseBody   *string
```

Comment:
add a newline separating the start of the private fields please

Also, what are these fields here for?